src/noit_check.c

```diff
@@ -1967,12 +1967,23 @@ noit_stats_set_metric(noit_check_t *check,
   memset(m, 0, sizeof(*m));
   if(noit_stats_populate_metric(m, name, type, value)) {
     mtev_memory_safe_free(m);
-    return;
+    return NULL;
   }
   noit_check_metric_count_add(1);
   c = noit_check_get_stats_inprogress(check);
   check_stats_set_metric_hook_invoke(check, c, m);
   __stats_add_metric(c, m);
+  return c;
+}
+void
+noit_stats_set_metric_timed(noit_check_t *check,
+                      const char *name, metric_type_t type,
+                      const void *value, const struct timeval *time) {
+  stats_t *c;
+  c = noit_stats_set_metric(check, name, type, value);
+  if(c) {
+    noit_check_stats_whence(c, time);
+  }
```

Comment:
This function sets the time for the whole stats structure, not just the metric.  I'm not sure we want to support such operations.  What's the expected use case where a caller will set the time for a "metric" but be fine with all metrics in the stats structure having their container's timestamp set?